corehq/apps/locations/fixtures.py

```diff
@@ -166,17 +167,15 @@ def _valid_parent_type(location):
 
 
 def _append_children(node, location_db, locations):
-    by_type = _group_by_type(locations)
-    for type, locs in by_type.items():
+    for type, locs in _group_by_type(locations):
         locs = sorted(locs, key=lambda loc: loc.name)
         node.append(_types_to_fixture(location_db, type, locs))
 
 
 def _group_by_type(locations):
-    by_type = defaultdict(lambda: [])
-    for loc in locations:
-        by_type[loc.location_type].append(loc)
-    return by_type
+    key = lambda loc: (loc.location_type.code, loc.location_type)
+    for (code, type), locs in groupby(sorted(locations, key=key), key=key):
```

Comment:
Kinda sad to be sorting these in production code since it incurs a performance penalty and ordering appears to not matter. However, `locations` is a set, so unless the tests are rewritten (not trivial) this is the easiest fix for now. Guessing the [sorting above](https://github.com/dimagi/commcare-hq/pull/12489/files#diff-29ea9598cea171ab08caff56a3506a63R171) is for the same reason.